### ISSUE TITLE
User controlled task priority

### ISSUE THREAD

**Issue by rhattersley** [blaze/dask#349]
When the minimise-work-in-progress heuristic has multiple tasks available with equal priority it would be useful to be able to steer the next task using a user specified priority. (e.g. When the number of final results is greater than the number of workers but the tasks sizes vary considerably, we can schedule work in reverse task size order to get a more even task distribution.)

This would go nicely with the visualisation mentioned in #346.

**Comment by mrocklin** [comment]
The state of the current async scheduler is such that this would be similar to but probably separate from the work in #346.  In previous versions there actually used to be a user-definable `score` function that could be used to select the next task.  This was removed as we changed to the last-in-first-out policy which we hard-coded with a stack.

A first pass at this would probably be to fork the existing async code and replace the stack with a priority queue, the key of which was defined by a user definable function (what would this take as input?)  This would require learning a bit about the state managed by the async scheduler, which is non-trivial but heavily documented in-line.

I'm unlikely to get to this before SciPy but would be happy to provide support.
